### PR TITLE
Add on-device compression

### PR DESCRIPTION
**Pull request by SoumyBhow**
Through the use of [CHDMAN](https://github.com/CharlesThobe/chdman) and [Maxcso](https://github.com/unknownbrackets/maxcso) along with extra packages in Termux, it's possible to compress ROMs without a PC (if we're scraping without a PC we might as well compress the roms too). I've added a compress.sh but it isn't working as of yet. Still, I figured you may be interested in it so I'm making the pull request to the dev branch as to not bork the main script just in case you want to tweak it. I hope this doesn't cause any problems....

Edit: I forgot to mention that PSP is also not in the compress script as of now since it uses maxcso instead of CHDMAN, so I couldn't figure out how to add it to the same whiptail menu. My apologies.

### CODE REVIEW

**Comment by dollerbill** [comment]
I really like this idea for someone that doesn't have convenient access to a PC, but I wonder what kind of drain running chdman would have on a lower-end system like an RP2. Were you able to test the on-device compression at all to see what kind of performance it yields? 


Also, a few files have artifacts left over from a rebase, they look like:

```git
<<<<<<< HEAD
	/bin/bash ~/compress.sh
=======
	/bin/bash ~/undo.sh
>>>>>>> bcd282dc1228bc72269485638e1f1065ddca9e5a
```
where the `<<<<<...`, `=======`, and `>>>>> ....` are breaks showing where the current HEAD and incoming changes differ. You can delete these by hand or reset these files back to the state before the commit and re-add your changes

**Comment by SoumyBhow** [comment]
I don't have a low end device but I did test chdman and maxcso on my OnePlus 7 Pro and both seem to do fine. Of course the time it takes depends on the size of the ROM you're compressing. As for the artifacts I'll look into them in the morning but thanks for letting me know.

**Comment by SoumyBhow** [comment]
> ```
> <<<<<<< HEAD
> 	/bin/bash ~/compress.sh
> =======
> 	/bin/bash ~/undo.sh
> >>>>>>> bcd282dc1228bc72269485638e1f1065ddca9e5a
> ```
> 
> where the `<<<<<...`, `=======`, and `>>>>> ....` are breaks showing where the current HEAD and incoming changes differ. You can delete these by hand or reset these files back to the state before the commit and re-add your changes

I'm trying to look into this but I don't understand what you mean exactly. On my side, it seems like it's fine after I check in ```git status```. Would you mind clarifying this artifact issue a bit more? I'm still new to all of this so please be patient with me.

**Comment by dollerbill** [comment]
@SoumyBhow of course, we all learn somewhere!


Screen Shot 2022-01-26 at 4 11 27 PM

If you look at this screenshot of your `startup.sh` file, you can see some of those artifacts I mentioned. 
Specifically, lines 15 and 23 are both unwanted changes and are leftover from merge conflicts.

[This article](https://phoenixnap.com/kb/how-to-resolve-merge-conflicts-in-git) can probably explain it better than I can, but the basics are you want to remove any of these indicators so they aren't tried to be interpreted when the scripts are run

**Comment by SoumyBhow** [comment]
> @SoumyBhow of course, we all learn somewhere!
> 
> Screen Shot 2022-01-26 at 4 11 27 PM
> 
> If you look at this screenshot of your `startup.sh` file, you can see some of those artifacts I mentioned. Specifically, lines 15 and 23 are both unwanted changes and are leftover from merge conflicts.
> 
> [This article](https://phoenixnap.com/kb/how-to-resolve-merge-conflicts-in-git) can probably explain it better than I can, but the basics are you want to remove any of these indicators so they aren't tried to be interpreted when the scripts are run

Ah okay the artifact was in the file itself. Thanks for the pointer! I was able to pull the latest update of the file and re-add the changes. I'll see if I can fix the script further.

File: setup.sh [hunk 1]
```diff
@@ -264,9 +283,34 @@ echo -ne  "Installing ${RED}Pegasus${NONE}..."
 #Launch Pegasus
 xdg-open ~/dragoonDoriseTools/pegasus-fe_alpha15-85-gfff1a5b2_android.apk
 echo -e  "${GREEN}OK${NONE}"
+<<<<<<< HEAD
```
Comment (dollerbill) [inline setup.sh]: One more here

File: setup.sh [hunk 2]
```diff
@@ -264,9 +283,34 @@ echo -ne  "Installing ${RED}Pegasus${NONE}..."
...
+bash scrap.sh
+clear
+echo -e  "${GREEN}Done!${NONE}"
+
+echo -e  "${STRONG}If you want to scrape more roms, update or uninstall Pegasus Installer:${NONE}"
+echo -e  "In Pegasus go to the Android Collection and look for Rom Scraper"
+
+
+echo -e "${RED}INSTALL EMULATORS${NONE} Remember to go to Retroarch's Main Menu -> Load Core -> Install or Restore a Core"
+=======
```
Comment (dollerbill) [inline setup.sh]: and here

File: setup.sh [hunk 3]
```diff
@@ -264,9 +283,34 @@ echo -ne  "Installing ${RED}Pegasus${NONE}..."
...
 echo ""
 echo -e "${YELLOW}Retroarch Cores${NONE}"
 echo -e "Remember to go to Retroarch's Main Menu -> Load Core -> Install or Restore a Core"
+>>>>>>> bcd282dc1228bc72269485638e1f1065ddca9e5a
```
Comment (dollerbill) [inline setup.sh]: last one!

**Comment by dollerbill** [comment]
> Ah okay the artifact was in the file itself. Thanks for the pointer! I was able to pull the latest update of the file and re-add the changes. I'll see if I can fix the script further.

the `startup` file looks good! I just saw a few more in `setup` but then it good to me

**Comment by SoumyBhow** [comment]
Ah shoot I missed the other files. I'll look into them as soon as I can. Thanks so much for your help @dollerbill

**Comment by SoumyBhow** [comment]
Okay so the artifacts *should* be all good now. All there's left is the actual compress script and adding PSP in the list.

**Comment by dragoonDorise** [comment]
Hey, this looks great, thanks! As soon as I can test that it works fine I’ll accept the PR

**Comment by SoumyBhow** [comment]
Sorry for being unresponsive. School's recently started up again. I'm having some trouble with the script I wrote. Namely there's a \r error that's bothering me. I have no idea why. Would anyone mind helping me with this?

**Comment by dollerbill** [comment]
@SoumyBhow I can try and take a look, can you share a screenshot or a copy of the error message that you're seeing and where you're seeing it?

**Comment by SoumyBhow** [comment]
Thanks for offering to help @dollerbill. I know I should be solving this myself but I just can't seem to wrap my head around it.

Screenshot_pegasus-android-metadata : bash_1

**Comment by SoumyBhow** [comment]
Okay so turns out the aforementioned error was cause I wrote the script on Windows. Now I just gotta fix the rest of the script...

**Comment by SoumyBhow** [comment]
Closing this cause I clearly don't have the time or skill to make this work. Sorry everyone

**Pull request closed without merging**